core_backend/app/auth/dependencies.py

```diff
@@ -9,62 +9,56 @@
 )
 from jose import JWTError, jwt
 
-from .config import (
-    ACCESS_TOKEN_EXPIRE_MINUTES,
-    JWT_ALGORITHM,
-    JWT_SECRET,
-    USER1_PASSWORD,
-    USER1_RETRIEVAL_KEY,
-    USER1_USERNAME,
-    USER2_PASSWORD,
-    USER2_RETRIEVAL_KEY,
-    USER2_USERNAME,
-)
-from .schemas import AccessLevel, AuthenticatedUser
+from ..database import get_async_session_direct
+from ..users.models import UserDB, get_user_by_token, get_user_by_username
+from ..utils import get_key_hash, setup_logger
+from .config import ACCESS_TOKEN_EXPIRE_MINUTES, JWT_ALGORITHM, JWT_SECRET
+from .schemas import AuthenticatedUser
+
+logger = setup_logger()
 
 bearer = HTTPBearer()
 oauth2_scheme = OAuth2PasswordBearer(tokenUrl="login")
 
 
-USERS = {
-    USER1_USERNAME: {
-        "password": USER1_PASSWORD,
-        "access_level": "fullaccess",
-    },
-    USER2_USERNAME: {
-        "password": USER2_PASSWORD,
-        "access_level": "fullaccess",
-    },
-}
-
-
-def auth_bearer_token(
+async def auth_bearer_token(
     credentials: HTTPAuthorizationCredentials = Depends(bearer),
-) -> str:
+) -> UserDB:
```

Comment:
Now returns the DB as we discussed, and so each endpoint doesnt have to its own check. If auth succeeds, it means that the user exists and we're good to go.